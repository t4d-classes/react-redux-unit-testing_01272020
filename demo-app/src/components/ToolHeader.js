import React from 'react';
import PropTypes from 'prop-types';

export const ToolHeader = ({ headerText }) => {

  // return React.createElement('header', null, React.createElement('h1', null, headerText));

  return <header>
    <h1>{headerText}</h1>
  </header>;


};

ToolHeader.propTypes = {
  headerText: PropTypes.string.isRequired,
};
