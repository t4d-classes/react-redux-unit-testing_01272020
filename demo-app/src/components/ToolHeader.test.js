import React from 'react';
import { render } from 'enzyme';

import { ToolHeader } from './ToolHeader';

describe('ToolHeader Snapshot Tests', () => {

  test('ToolHeader displays Hello World', () => {

    const toolHeaderHtml = render(
      <ToolHeader headerText="Hello World!" />
    ).html()

    const component = JSON.stringify(toolHeaderHtml);

    expect(component).toMatchSnapshot();

  });

});