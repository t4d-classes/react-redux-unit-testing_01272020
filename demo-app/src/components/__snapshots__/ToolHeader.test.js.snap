// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolHeader Snapshot Tests ToolHeader displays Hello World 1`] = `"\\"<h1>Hello World!</h1>\\""`;
